### PR TITLE
7839 refactor secondary contacts into petitioners array

### PR DESCRIPTION
**Pull request by mrinsin**

### CODE REVIEW

File: shared/src/business/entities/EntityValidationConstants.js [hunk 1]
```diff
@@ -420,6 +420,8 @@ const WORK_ITEM_VALIDATION_RULE_KEYS = {
   workItemId: JoiValidationConstants.UUID.required(),
 };
 
+const PETITIONER_VALIDATION_RULE_KEYS = {};
```
Comment (mrinsin) [inline shared/src/business/entities/EntityValidationConstants.js]: Remove

File: shared/src/business/entities/cases/Case.js [hunk 2]
```diff
@@ -634,6 +632,8 @@ Case.VALIDATION_RULES = {
     },
   ).description('When the case fee was waived.'),
   // Individual items are validated by the ContactFactory.
+
+  // items is a function that accepts contactType and returns ContactFactory.getValidationRules(CONTACT_TyPE)
```
Comment (mrinsin) [inline shared/src/business/entities/cases/Case.js]: remove

File: shared/src/business/entities/PrivatePractitioner.test.js [hunk 3]
```diff
@@ -271,10 +273,25 @@ describe('PrivatePractitioner', () => {
...
+            ...MOCK_CASE,
+            partyType: PARTY_TYPES.petitionerSpouse,
+            petitioners: [
+              {
+                ...getContactPrimary(MOCK_CASE),
+                contactId: '152732ec-4d31-4b9b-925b-2b746d9fbf08',
+                contactType: CONTACT_TYPES.primary,
+              },
+              {
+                ...getContactPrimary(MOCK_CASE),
```
Comment (rosiehsmith) [inline shared/src/business/entities/PrivatePractitioner.test.js:287]: getContactSecondary instead?

File: shared/src/business/entities/cases/PublicContact.js [hunk 4]
```diff
@@ -25,7 +26,9 @@ PublicContact.prototype.init = function init(rawContact) {
 };
 
 PublicContact.VALIDATION_RULES = joi.object().keys({
-  contactType: JoiValidationConstants.STRING.optional(),
+  contactType: JoiValidationConstants.STRING.valid(
+    ...Object.values(CONTACT_TYPES),
+  ).optional(),
```
Comment (rosiehsmith) [inline shared/src/business/entities/cases/PublicContact.js:31]: keep this optional, but make it required in ContactFactory

File: shared/src/business/useCases/updatePetitionerInformationInteractor.js [hunk 5]
```diff
@@ -298,50 +298,75 @@ exports.updatePetitionerInformationInteractor = async (
...
+      sealedAndUnavailable: oldCaseContactSecondary?.sealedAndUnavailable,
+      secondaryName: oldCaseContactSecondary?.secondaryName,
+      title: oldCaseContactSecondary?.title,
+      ...secondaryEditableFields,
+    });
+  } catch (e) {
+    applicationContext.logger.info(
+      'no contact secondary was found on the case',
+    );
+    //TODO handle ADDING contactSecondary when none was previously on the case?
```
Comment (rachelschneiderman) [inline shared/src/business/useCases/updatePetitionerInformationInteractor.js]: check if this already works, if not, fixit!

File: shared/src/business/useCases/updatePetitionerInformationInteractor.js [hunk 6]
```diff
@@ -298,50 +298,75 @@ exports.updatePetitionerInformationInteractor = async (
...
+      email: oldCaseContactSecondary?.email,
+      hasEAccess: oldCaseContactSecondary?.hasEAccess,
+      isAddressSealed: oldCaseContactSecondary?.isAddressSealed,
+      sealedAndUnavailable: oldCaseContactSecondary?.sealedAndUnavailable,
+      secondaryName: oldCaseContactSecondary?.secondaryName,
+      title: oldCaseContactSecondary?.title,
+      ...secondaryEditableFields,
+    });
+  } catch (e) {
+    applicationContext.logger.info(
```
Comment (rachelschneiderman) [inline shared/src/business/useCases/updatePetitionerInformationInteractor.js]: clean up a bit? this isn't really an error, some cases do not have contact secondary-s

File: shared/src/business/useCases/updateSecondaryContactInteractor.js [hunk 7]
```diff
@@ -52,13 +53,22 @@ exports.updateSecondaryContactInteractor = async (
...
+
+  const oldContactSecondary = cloneDeep(caseEntity.getContactSecondary());
+
+  const updatedSecondaryContact = {
+    ...oldContactSecondary,
+    ...editableFields,
+  };
+
+  try {
+    caseEntity.updatePetitioner(updatedSecondaryContact);
```
Comment (rachelschneiderman) [inline shared/src/business/useCases/updateSecondaryContactInteractor.js:66]: look at everywhere we call updatePetitioner(), make sure we send that case back through the case constructor in order to validate the contacts again

File: web-client/src/presenter/actions/ManualAssociation/setEditPrivatePractitionersAction.js [hunk 8]
```diff
@@ -11,19 +10,28 @@ import { state } from 'cerebral';
  * @param {object} providers.store the cerebral store object
  * @returns {Promise<*>} the promise of the completed action
  */
-export const setEditPrivatePractitionersAction = async ({ get, store }) => {
+export const setEditPrivatePractitionersAction = async ({
```
Comment (rachelschneiderman) [inline web-client/src/presenter/actions/ManualAssociation/setEditPrivatePractitionersAction.js:13]: jsdocs

File: web-client/src/views/FilingPartiesForm.jsx [hunk 9]
```diff
@@ -49,7 +47,7 @@ export const FilingPartiesForm = connect(
               className="usa-checkbox__label inline-block"
               htmlFor="party-primary"
             >
-              {caseDetailContactPrimaryHelper.contactPrimary.name}
+              {caseDetailContactHelper.contactPrimary.name}
```
Comment (rachelschneiderman) [inline web-client/src/views/FilingPartiesForm.jsx:50]: remove `contactSecondary` from elasticsearch query source fields

**Review by mrinsin — COMMENTED** [review]
Remember to update mock cases to not have contact primary and secondary

**Review by rachelschneiderman — APPROVED** [review]

**Pull request merged**